viewing_party/main.py

```diff
@@ -0,0 +1,195 @@
+    return mean 
+
+
+# test 12 funct collects the genre data and returns most watched genre 
+
+# test 13 make sure the function returns none if list is empty 
+def get_most_watched_genre(user_data):
+    word_count = {}
+    if user_data['watched'] == []:
+        return None
+    else:
+        for movie_dict in user_data['watched']:
+            if movie_dict['genre'] not in word_count:
+                word_count[movie_dict['genre']] = 1
+            else:
+                word_count[movie_dict['genre']] += 1
+        max_genre = max(word_count)
+        return max_genre
+
+
+
+#   ***** wave 3 tests 14-18 **** 
+# test 14 funct get_unique_watched takes in user data 
+# returns list of movies in user data that are not in friends data
+
+# test 15 returns an empty list if everyone watched the same movies 
+
+def get_unique_watched(user_data):
+
+    friends_watched = []
+    for dictionary in user_data['friends']:
+        for title_list in dictionary.values():
+            for dict in title_list:
+                friends_watched.append(dict)
+
+
+    unique_watched = [movie for movie in user_data['watched']\
+         if movie not in friends_watched]
+
+    # for movie_title_dict in user_data['watched']:
+    #     if movie_title_dict not in friends_watched:
+    #         unique_watched.append(movie_title_dict)
+
+    return unique_watched
+
+
+
+
+
+# test 16 function get_friends_unique_watched
+# takes in user_data determines which movies at least one of 
+# users friend has watched, but user has not.  return as 
+# list of dict. 
+
+# test 17 does it work with only one friend
+
+
+#  test 18 make sure the function returns [] if the user has 
+# watched all the movies friends have watched. 
+
+
+def get_friends_unique_watched(user_data): 
+    friends_watched = []
+# creates a list of dict of movies friends have watched 
+    for friend_dict in user_data['friends']:
+        for title_list in friend_dict.values():
+            for title in title_list:
+                friends_watched.append(title)
+# set a new list to remove duplicate movies 
+    friends_watched_remove_dupes = []
+    for movie in friends_watched:
+        if movie not in friends_watched_remove_dupes:
+            friends_watched_remove_dupes.append(movie)
+
+    more_unique = [movie for movie in friends_watched_remove_dupes 
+if movie not in user_data['watched']]
+
+    return more_unique
+
+
+# ****** wave 4 tests 19 - 20 ******** 
+# test 19 funct get_available_recs takes in user_data 
+# w/ dict key "subscriptions" value is list strings 
+# each friends watched movie data will also have a host key, whose 
+# value will be the streaming service as string 
+# output should be a list of reccomended moves that 
+#   a: user has not watched 
+#   b: at least one friend has watched 
+#   c: the host of the movie is in the users "subscriptions"
+
+# test 20 ensure you return an empty list if there are no intersections 
+def get_available_recs(user_data):
```

Comment:
👍 , This has a pretty deep list comprehension.  It's good practice, but be a little careful if the logic is getting too twisted for readability's sake.